doboto/Endpoint.py

```diff
@@ -80,3 +80,51 @@ def pages(self, request_url, expect, params=None):
                 next_url = None
 
         return items
+
+    def action_result(self, action, wait, poll, timeout):
+        """
+        General action result processor for waiting
+        """
+
+        start_time = time.time()
+
+        while wait and action["status"] == "in-progress":
+
+            time.sleep(poll)
+            try:
+                action = self.do.action.info(action["id"])
+            except:
+                pass
+
+            if time.time() - start_time > timeout:
+                raise DOBOTOException("Timeout on polling", action)
+
+        return action
+
+    def actions_result(self, actions, wait, poll, timeout):
+        """
+        General actions result processor for waiting
+        """
+
+        start_time = time.time()
+
+        info = [index for index, action in enumerate(actions)
+                if action["status"] == "in-progress"]
+
+        while wait and len(info) > 0:
```

Comment:
Nothing in this loop appears to update the value of wait?  Is it useful as a conditional for the loop?  if the point is to only loop if wait is set to a non-False value, it would be a bit more readable imho to just throw in something like:

```python
if not wait:
    return actions
```